src/common/temporal/calendar-date.ts

```
@@ -160,13 +164,19 @@ export class CalendarDate<IsValid extends boolean = DefaultValidity>
   static max(): undefined;
   static max(...dateTimes: DateTime[]): CalendarDate;
   static max(...dateTimes: DateTime[]): CalendarDate | undefined {
-    return CalendarDate.fromDateTime(super.max(...dateTimes));
+    if (dateTimes.length === 0) return undefined;
+    return CalendarDate.fromDateTime(
+      super.max(...(dateTimes as unknown as NonEmptyArray<DateTime>)),
```

Biome (1.9.4)

[error] Using super in a static context can be confusing.

super refers to a parent class.

(lint/complexity/noThisInStatic)